### PR TITLE
Updating documentation; adding content checking

### PR DESCRIPTION
**Pull request by tpmccallum**
Signed-off-by: tpmccallum <tim.mccallum@fermyon.com>

Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [x] The `title, `template`, and `date` are all set
- [x] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result) 
- [x] Has passed [`bart check`](https://bartholomew.fermyon.dev/quickstart)
- [x] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)
- [x] Headings are using Title Case
- [x] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [x] Have tested with `npm run test` and resolved all errors

### CODE REVIEW

File: content/cloud/contributing.md [hunk 1]
```diff
@@ -60,24 +63,66 @@ Create a new remote for the upstream (a pointer to the original repository to wh
...
+Once your changes have been checked, go ahead and add your changes by moving to a top-level directory, under which your changes exist i.e. `cd ~/developer`.
 
 Add your changes using the following command:
 
 <!-- @selectiveCpy -->
 
 ```bash
+$ cd ~
+$ cd developer
 $ git add
```
Comment (karthik2804) [inline content/cloud/contributing.md:115]: as above

File: content/cloud/contributing.md [hunk 2]
```diff
@@ -60,24 +63,66 @@ Create a new remote for the upstream (a pointer to the original repository to wh
...
 
 Before committing, please ensure that your GitHub installation is configured sufficiently so that you can `--signoff` as part of the `git commit` command. For example, please ensure that the `user.name` and `user.email` are configured in your terminal. You can check if these are set by typing `git config --list`.
 
 If you need to set these values please use the following commands:
 
+<!-- @selectiveCpy -->
+
 ```bash
 $ git config user.name "yourusername"
 $ git config user.email "youremail@somemail.com"
```
Comment (karthik2804) [inline content/cloud/contributing.md:128]: Might want to split this into 2 code blocks as the user needs to modify each of the statements.

File: content/cloud/contributing.md [hunk 3]
```diff
@@ -60,24 +63,66 @@ Create a new remote for the upstream (a pointer to the original repository to wh
...
+
+```bash
+$ echo "hello"
+hello
+```
+
+> Note: If the command, that starts with `$`, is deliberately spread over two lines (by escaping the newline character), then the copy mechanism will still copy the second line which is technically still part of that single command.
+
+The no copy annotation (`<!-- @nocpy -->`) preceeds a code block where no copy and pasting of code is intended. For example, use no copy when just displaying output or information to demonstrate a terminal's output.
+
```
Comment (karthik2804) [inline content/cloud/contributing.md:79]: I suggest adding the language tag of `text` to any code blocks which warrant the `<!-- @nocpy -->` as well.

File: content/cloud/contributing.md [hunk 4]
```diff
@@ -60,24 +63,66 @@ Create a new remote for the upstream (a pointer to the original repository to wh
 $ git remote add upstream https://github.com/fermyon/developer
 ```
 
-## 5. Add Changes
+## 5. Code Blocks and Annotations
+
+It is highly recommended that you use either the `<!-- @selectiveCpy -->` or the `<!-- @nocpy -->` annotation before each of your code blocks, and that each code block defines the appropriate [syntax highlighting](https://rdmd.readme.io/docs/code-blocks#language-support).
```
Comment (karthik2804) [inline content/cloud/contributing.md:67]: The annotation can be skipped for code blocks with example code snippets (i.e) non terminal/output type codeblocks

**Review by karthik2804 — CHANGES_REQUESTED** [review]

File: content/cloud/contributing.md [hunk 5]
```diff
@@ -60,24 +63,66 @@ Create a new remote for the upstream (a pointer to the original repository to wh
...
+```
+With all Node dependencies installed, you can now check for broken links and also lint your markdown files using the following `test` command:
+
+<!-- @selectiveCpy -->
 
-Once you are satisfied with your contribution go ahead and add your changes by moving to a top-level directory, under which your changes exist i.e. `cd ~/developer`.
+```bash
+$ cd ~
+$ cd developer
+$ npm test
```
[the remark this answers is not in the document]
Comment (tpmccallum) [reply content/cloud/contributing.md:101]: Thanks @karthik2804 I have updated this in the latest commit.

File: content/cloud/contributing.md [hunk 6]
```diff
@@ -60,24 +63,66 @@ Create a new remote for the upstream (a pointer to the original repository to wh
...
+Once you are satisfied with your contribution, you can programmatically check your content.
+
+If you have not done so already, please go ahead and perform the `npm install` command; to enable Node dependencies such as `markdownlint-cli2`.
+
+<!-- @selectiveCpy -->
+
+```bash
+$ cd ~
+$ cd developer
+$ npm install
```
[the remark this answers is not in the document]
Comment (tpmccallum) [reply content/cloud/contributing.md:91]: Thanks.

File: content/cloud/contributing.md (hunk 1 above)
Comment (tpmccallum) [reply content/cloud/contributing.md:115]: Thanks.

File: content/cloud/contributing.md (hunk 3 above)
Comment (tpmccallum) [reply content/cloud/contributing.md:79]: Done, thanks.

File: content/cloud/contributing.md (hunk 4 above)
Comment (tpmccallum) [reply content/cloud/contributing.md:67]: Also, updated in latest commit.

File: content/cloud/contributing.md (hunk 2 above)
Comment (tpmccallum) [reply content/cloud/contributing.md:128]: Done.

**Review by karthik2804 — COMMENTED** [review]
LGTM!

**Pull request merged**